carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/application/RecruitmentServiceHost.kt

```diff
@@ -169,12 +231,15 @@ class RecruitmentServiceHost(
         val recruitment: Recruitment = getRecruitmentOrThrow( studyId )
 
         // Get study deployment status list.
-        val studyDeploymentIds = recruitment.participantGroups.keys
+        val deployedStudyDeploymentIds = recruitment.participantGroups.filter { it.value.isDeployed }.keys
+        val stagedParticipantGroups = recruitment.participantGroups.filter { !it.value.isDeployed }.values.map {
+            getStagedParticipantGroupStatus( it, recruitment )
+        }
         val studyDeploymentStatusList: List<StudyDeploymentStatus> =
-            if ( studyDeploymentIds.isEmpty() ) emptyList()
-            else deploymentService.getStudyDeploymentStatusList( studyDeploymentIds )
+            if ( deployedStudyDeploymentIds.isEmpty() ) emptyList()
+            else deploymentService.getStudyDeploymentStatusList( deployedStudyDeploymentIds )
 
-        return studyDeploymentStatusList.map { recruitment.getParticipantGroupStatus( it ) }
+        return stagedParticipantGroups + studyDeploymentStatusList.map { recruitment.getParticipantGroupStatus( it ) }
```

Comment:
I was wondering why you updated this. But, I see it is because you also return _staged_ participants now. Make sure to update the documentation (it mentiones "all deployed participant groups"). I guess, since before you couldn't have just "staged" participants, we can still consider this fully backwards compatible.

You can make this code more readable by not intermingling getting the status of staged groups with that of deployed groups. They could be two separate "code blocks" with a comment above.